src/com/stuypulse/stuylib/math/interpolation/NearestInterpolator.java

```diff
@@ -0,0 +1,73 @@
+/* Copyright (c) 2022 StuyPulse Robotics. All rights reserved. */
+/* This work is licensed under the terms of the MIT license */
+/* found in the root directory of this project. */
+
+package com.stuypulse.stuylib.math.interpolation;
+
+import com.stuypulse.stuylib.math.SLMath;
+import com.stuypulse.stuylib.math.Vector2D;
+
+/**
+ * This class is a modified version of Linear Interpolation (Lerp). Using this class will allow you to interpolate
+ * between the NEAREST two reference points, as opposed to just interpolating between only two points in Linear Interpolator.
+ * @author Eric Lin (ericlin071906@gmail.com, Github: cire694)
+ * @author Sam (sam.belliveau@gmail.com)
+ */
+
+public class NearestInterpolator implements Interpolator {
+
+    private final Vector2D[] points; 
+
+    // sorts the points in ascending order
+    public NearestInterpolator(Vector2D... points) {
+        this.points = Interpolator.getSortedPoints(points); 
+    }
+    
+    @Override
+    /**
+     * @param x the point of extrapolation to get an output
+     * @return interpolated value
+     */
+    public double interpolate(double x) {
+        // this section will find the nearest refernce points to the distance
+        Vector2D left = Vector2D.kOrigin; // points are 0, 0 for kOrigin
+        Vector2D right = Vector2D.kOrigin; 
+
+        // searching for the points on the left and right of the target point. 
+        for (int i = 1; i < points.length; i ++){ 
+            Vector2D left_temp = points[i - 1];
+            Vector2D right_temp = points[i - 0];
+
+            if (left_temp.x <= x && x <= right_temp.x){
+                left = left_temp;
+                right = right_temp;
+                
+                break;
+            }
```

Comment:
if `x` is before or after your range of x values, then it doesn't find points and it uses `kOrigin` for both points. basically it goes to 0. 

is this something you want? in this case it could alternatively use the 2 outer values and use those to interpolate outside the range. make sure you graph this